rfcs/0181-list-index-syntax.md

```diff
@@ -0,0 +1,290 @@
+---
+feature: list-index-syntax
```

Comment:
Since we are here, what about extending this to sets?

```nix
{ x=1; y=2; }.["x"]
```

Reply:
Why? You can already write `{ x = 1; y = 2; }."x"`.

Reply:
I think this is a good argument why the `.0` syntax is more consistent and should be preferred.

Reply:
> On the other hand, in guise of speculation, with the bracketed syntax we can have nice things like
>
> `.[ "x" "y" ]`

I think this should be discussed in its own RFC.

IMO `.int` (e.g. `.2`) is consistent with the attr indexing syntax `."name"`.

Since attr names can't be int literals, this fits in nicely.

`.${expression}` indexing will cover more complex scenarios too.

This would leave `.[/*other syntax*/]` available for any future use, to be tackled by another RFC.